nixos/tests/nvidia-container-toolkit.nix

```diff
@@ -0,0 +1,158 @@
+import ./make-test-python.nix (
+  {
+    pkgs,
+    lib,
+    system,
+    ...
+  }:
+  let
+    unfreeAndInsecurePkgs = import ../.. {
+      inherit system;
+      config = {
+        allowUnfree = true;
+        permittedInsecurePackages = [ "openssl-1.1.1w" ];
+      };
+    };
+    testContainerImage =
+      let
+        testCDIScript = pkgs.writeShellScriptBin "test-cdi" ''
+          die() {
+            echo "$1"
+            exit 1
+          }
+
+          check_file_referential_integrity() {
+            echo "checking $file referential integrity"
+            files=$(set -o pipefail && \
+                    ${pkgs.glibc.bin}/bin/ldd "$1" | \
+                    ${pkgs.gnugrep}/bin/grep '=>' | \
+                    ${pkgs.gnused}/bin/sed "s/.* => //" | \
+                    ${pkgs.gnused}/bin/sed "s/ (.*//") || exit 1
+
+            for file in $files; do
+              if [ ! -f "$file" ]; then
+                die "$file does not exist in the container filesystem"
+              fi
+            done
+          }
+
+          check_directory_referential_integrity() {
+            ${pkgs.findutils}/bin/find "$1" -type f -print0 | while read -d $'\0' file; do
+              if [[ $(${pkgs.file}/bin/file "$file" | ${pkgs.gnugrep}/bin/grep ELF) ]]; then
+                check_file_referential_integrity "$file" || exit 1
+              else
+                echo "skipping $file"
+              fi
+            done
+          }
+
+          check_directory_referential_integrity "/usr/bin" || exit 1
+          check_directory_referential_integrity "${pkgs.addDriverRunpath.driverLink}" || exit 1
+          check_directory_referential_integrity "/usr/local/nvidia" || exit 1
+        '';
+      in
+      pkgs.dockerTools.buildImage {
+        name = "cdi-test";
+        tag = "latest";
+        config = {
+          Cmd = [ "${testCDIScript}/bin/test-cdi" ];
+          Env = [
+            "LD_LIBRARY_PATH=${unfreeAndInsecurePkgs.linuxPackages.nvidia_x11}/lib:${lib.getLib unfreeAndInsecurePkgs.openssl_1_1}/lib"
```

Comment:
Getting the unfree stuff into the CI is tricky, but thankfully there's no need: you can build the open driver instead